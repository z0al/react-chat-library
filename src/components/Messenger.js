// @ts-check

// Packages
import React from 'react';

// Ours
import { prepare } from '../utils';
import MessageList from './MessageList';
import Composer from './Composer';

/**
 * @typedef Props
 * @property {string} text
 * @property {Array} messages
 * @property {object} [user]
 * @property {object} [style]
 * @property {function} [onSend]
 * @property {function} [onTextChange]
 * @augments {React.Component<Props>}
 */
class Messenger extends React.Component {
	/** @type Props */
	constructor(props) {
		super(props);

		this.state = { text: props.text || '' };
	}

	onSubmit = () => {
		if (this.props.onSend) {
			this.props.onSend(this.state.text);
			this.setState({ text: '' });
		}
	};

	onTextChange = text => {
		this.setState({ text });

		this.props.onTextChange && this.props.onTextChange(text);
	};

	render() {
		return (
			<div style={this.props.style}>
				<MessageList {...prepare(this.props)} />
				<Composer
					{...prepare(this.props)}
					text={this.state.text}
					onSubmit={this.onSubmit}
					onTextChange={this.onTextChange}
				/>
				<style jsx>{`
					div {
						display: flex;
						flex: 1;
						flex-direction: column;
						height: 600px;
					}
				`}</style>
			</div>
		);
	}
}

/** @type Props */
Messenger.defaultProps = {
	text: '',
	messages: [],
	style: {}
};

export default Messenger;
