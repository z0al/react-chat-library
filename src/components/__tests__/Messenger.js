import React from 'react';
import renderer from 'react-test-renderer';

import Messenger from '../Messenger';

test('should match the snapshot', () => {
	const tree = renderer.create(<Messenger />).toJSON();
	expect(tree).toMatchSnapshot();
});
