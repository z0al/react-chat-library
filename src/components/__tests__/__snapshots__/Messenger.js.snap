// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="jsx-3737022624"
  style={Object {}}
>
  <ul
    className="jsx-2992020076"
  />
  <div
    className="jsx-593056877"
    style={Object {}}
  >
    <div
      className="jsx-233852188"
      style={Object {}}
    >
      <textarea
        className="jsx-233852188"
        data-testid="text-input"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Type a message"
        rows={1}
        style={Object {}}
        value=""
        wrap="soft"
      />
    </div>
    <div
      className="jsx-3725816649"
      style={Object {}}
    >
      <button
        className="jsx-3725816649"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <span
          className="jsx-3725816649"
        >
          +
        </span>
      </button>
    </div>
  </div>
</div>
`;
